### ISSUE TITLE
rustdoc: private_doc_tests lint no longer fires on stable

### ISSUE THREAD

**Issue by ehuss** [rust-lang/rust#75951]
As of `rustc 1.47.0-beta.1`, the `private_doc_tests` lint is no longer working.

```rust
#![warn(private_doc_tests)]
/// hi!
/// ```
/// let x = 1;
/// ```
fn f() {}
```

I expected to see this happen: `cargo doc` should issue `warning: documentation test in private item`

Instead, this happened: No warnings emitted.

This appears to have changed in #74855.  I don't see any discussion there about this, so I'm guessing it was unintentional.  

cc @jyn514.

**Comment by jyn514** [comment]
This was definitely unintentional. I thought I added a test for this ...

**Comment by jyn514** [comment]
Hmm, it tests with `deny(rustdoc)` but not with `deny(missing_doc_code_examples)`. Maybe the _name_ of the lint wasn't registered?

**Comment by ehuss** [comment]
The issue is [this line](https://github.com/rust-lang/rust/blob/1f2dd3b56aec4d8f97eb0b0552c166ef549d3316/src/librustdoc/passes/doc_test_lints.rs#L103).  Compiletest runs with RUSTC_BOOTSTRAP set.  I don't recall off the top of my head how to force a test to be stable.

**Comment by jyn514** [comment]
Wait now I'm confused. I thought the point of this lint was that it only runs on nightly, because we're not sure if we want to stabilize it.

cc @rust-lang/rustdoc

**Comment by jyn514** [comment]
@ehuss are you sure this used to work?

**Comment by ehuss** [comment]
The `private_doc_tests` lint has been on stable since 1.32. It is documented [here](https://doc.rust-lang.org/rustdoc/lints.html?#private_doc_tests).

**Comment by jyn514** [comment]
Got it, I mixed it up with `missing_doc_code_examples`. Yeah this is a bug then - the two lints should be treated differently.

**Comment by jyn514** [comment]
Actually it looks like I introduced the opposite bug for `missing_doc_code_examples` - it now runs on beta when it shouldn't.

**Comment by spastorino** [comment]
Assigning `P-high` as [discussed as part of the Prioritization Working Group procedure](https://rust-lang.zulipchat.com/#narrow/stream/245100-t-compiler.2Fwg-prioritization.2Falerts/topic/I-prioritize.20.2375951.20rustdoc.3A.20private_doc_tests.20lint.20no.20long.E2.80.A6) and removing `I-prioritize`.
